sigmastate/src/main/scala/sigmastate/serialization/TupleSerializer.scala

```diff
@@ -24,7 +25,7 @@ case class TupleSerializer(cons: Seq[Value[SType]] => Value[SType])
 
   override def parse(r: SigmaByteReader): Value[SType] = {
     val size = r.getByte()
-    val values = ValueSerializer.newArray[SValue](size) // assume size > 0 so always create a new array
+    val values = safeNewArray[SValue](size) // assume size > 0 so always create a new array
```

Comment:
This refers to the fact that if the resulting array is empty, the allocation can be safely avoided because empty arrays are immutable and can be shared. See for example [here](https://github.com/ScorexFoundation/sigmastate-interpreter/blob/e1af1878fcb6ece28114d65b878cdc913c0e7bf3/sigmastate/src/main/scala/sigmastate/serialization/BlockValueSerializer.scala#L32)
This gives speedups observable in a profiler.

But since tuples now always have 2 elements, this array is always non-empty, so the optimization is not applicable.

clarified the comment.